src/control/lib/control/cont_test.go

```diff
@@ -0,0 +1,163 @@
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package control
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+
+	"github.com/daos-stack/daos/src/control/common"
+	mgmtpb "github.com/daos-stack/daos/src/control/common/proto/mgmt"
+	"github.com/daos-stack/daos/src/control/logging"
+)
+
+func TestControl_ContSetOwner(t *testing.T) {
+	testPoolUUID := uuid.New().String()
+	testContUUID := uuid.New().String()
+
+	validReq := &ContSetOwnerReq{
+		PoolUUID: testPoolUUID,
+		ContUUID: testContUUID,
+		User:     "someuser@",
+		Group:    "somegroup@",
+	}
+
+	for name, tc := range map[string]struct {
+		mic    *MockInvokerConfig
+		req    *ContSetOwnerReq
+		expErr error
+	}{
+		"nil request": {
+			req:    nil,
+			expErr: errors.New("nil request"),
+		},
+		"bad container UUID": {
+			req: &ContSetOwnerReq{
+				PoolUUID: testPoolUUID,
+				ContUUID: "junk",
+			},
+			expErr: errors.New("invalid UUID"),
+		},
+		"bad pool UUID": {
+			req: &ContSetOwnerReq{
+				PoolUUID: "garbage",
+				ContUUID: testContUUID,
+			},
+			expErr: errors.New("invalid UUID"),
+		},
+		"no container UUID": {
+			req: &ContSetOwnerReq{
+				PoolUUID: testPoolUUID,
+			},
+			expErr: errors.New("invalid UUID"),
+		},
+		"no pool UUID": {
+			req: &ContSetOwnerReq{
+				ContUUID: testContUUID,
+			},
+			expErr: errors.New("invalid UUID"),
+		},
+		"no user or group": {
+			req: &ContSetOwnerReq{
+				PoolUUID: testPoolUUID,
+				ContUUID: testContUUID,
+			},
+			expErr: errors.New("no user or group specified"),
+		},
+		"both user and group success": {
+			req: validReq,
+			mic: &MockInvokerConfig{
+				UnaryResponse: mockMSResponse("host1", nil,
+					&mgmtpb.ContSetOwnerResp{},
+				),
+			},
+		},
+		"user-only success": {
+			req: &ContSetOwnerReq{
+				PoolUUID: testPoolUUID,
+				ContUUID: testContUUID,
+				User:     "someuser@",
+			},
+			mic: &MockInvokerConfig{
+				UnaryResponse: mockMSResponse("host1", nil,
+					&mgmtpb.ContSetOwnerResp{},
+				),
+			},
+		},
+		"group-only success": {
+			req: &ContSetOwnerReq{
+				PoolUUID: testPoolUUID,
+				ContUUID: testContUUID,
+				Group:    "somegroup@",
+			},
+			mic: &MockInvokerConfig{
+				UnaryResponse: mockMSResponse("host1", nil,
+					&mgmtpb.ContSetOwnerResp{},
+				),
+			},
+		},
+		"local failure": {
+			req: validReq,
+			mic: &MockInvokerConfig{
+				UnaryError: errors.New("local failed"),
+			},
+			expErr: errors.New("local failed"),
+		},
+		"remote failure": {
+			req: validReq,
+			mic: &MockInvokerConfig{
+				UnaryResponse: mockMSResponse("host1", errors.New("remote failed"), nil),
+			},
+			expErr: errors.New("remote failed"),
+		},
+		"DAOS reported failure": {
+			req: validReq,
+			mic: &MockInvokerConfig{
+				UnaryResponse: mockMSResponse("host1", nil,
+					&mgmtpb.ContSetOwnerResp{Status: -1001},
+				),
+			},
+			expErr: errors.New("DAOS returned error code: -1001"),
+		},
```

Comment:
With the `getMSResponse()` helper, this kind of test case shouldn't be necessary. There are already unit tests for the interceptors and for the api helpers, so we should be able to keep these API surface unit tests pretty lean to just cover the validation and error handling inside their corresponding API function.

Reply:
Removed